### ISSUE TITLE
Wang Corner-Tiles support for Autotile

### ISSUE THREAD

**Issue by Maltavius** [godotengine/godot-proposals#2283]
### Describe the project you are working on
Top-Down 2D RPG game which would really benefit from having a possibility to specify multiple tile types in one autotile. So I think that adding the Wang tiles system to autotile could be really useful.

Wang corner tiles make for good terrain tiles
![image](https://user-images.githubusercontent.com/10133945/107852024-71a26e00-6e0e-11eb-9940-3a9fc023e917.png)

### Describe the problem or limitation you are having in your project
Godot currently only seem to support Blob tilesets.
Blob-tilesets are graphically intense to create.

This is an auto-generated Blob-tileset
![image](https://user-images.githubusercontent.com/10133945/107851932-a530c880-6e0d-11eb-81b4-1cfd407eeb79.png)

This is the same source-images but as a wang tileset
![image](https://user-images.githubusercontent.com/10133945/107851926-9d712400-6e0d-11eb-9b1c-4077ffa1787f.png)

### Describe the feature / enhancement and how it helps to overcome the problem or limitation
Edge matching Wang tiles tend to produce path or maze designs. A second type of tile set are corner tiles. These are matched by their corners and tend to produce patch or terrain designs.
http://cr31.co.uk/stagecast/wang/2corn.html

Variations of a tile could easily be produced and selected from the y-axis of the tileset. Like this example
![image](https://user-images.githubusercontent.com/10133945/107852005-33a54a00-6e0e-11eb-9e02-91c845dcf85a.png)

Se demos here:
http://cr31.co.uk/stagecast/scast/corner.html

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams
The link provides demos and diagrams
http://cr31.co.uk/stagecast/wang/2corn.html

### If this enhancement will not be used often, can it be worked around with a few lines of script?
No.

### Is there a reason why this should be core and not an add-on in the asset library?
It's a core type of tile-creating used for top-down 2D games.

**Comment by Calinou** [comment]
cc @groud

**Comment by Nekuromu** [comment]
This would definitely help, 2D RPGs seem to use Wang tiles often. Tilesetter can generate Wang tilesets very quickly & exports to Godot... bitmask data and all.

This feature would create great opportunities for quick RPG tile creation. Even the Docs on Tilesetter mention the scarcity of software that can do multiborder tilesets. Godot can be one of the few that does.

**Comment by Maltavius** [comment]
Any chance of Want-Tilesets being added to 4.0?

**Comment by Calinou** [comment]
@Maltavius The TileMap and TileSet editors have been rewritten for 4.0, so I recommend you try a [nightly build](https://hugo.pro/projects/godot-builds/) and see if the new editor suits your needs.

**Comment by Maltavius** [comment]
> @Maltavius The TileMap and TileSet editors have been rewritten for 4.0, so I recommend you try a [nightly build](https://hugo.pro/projects/godot-builds/) and see if the new editor suits your needs.
> 
> See the various progress reports (from oldest to newest):
> 
> * https://godotengine.org/article/tiles-editor-rework
> * https://godotengine.org/article/tiles-editor-progress-report-2
> * https://godotengine.org/article/tiles-editor-progress-3
> * https://godotengine.org/article/tiles-editor-progress-4
> * https://godotengine.org/article/tiles-editor-progress-report-5

Yeah that doesn't help much when I can't even get it to create a tilemap with the latest Nightly.
Also, there is nothing in those posts suggesting that Wang-Corner Tiles has been added to the supported list.

If it was, I suspect this Issue would have had a Milestone and someone added to it

**Comment by Zireael07** [comment]
Godot engine repo doesn't use milestones for issues, really.

And the tilemaps have indeed been rewritten completely, so the only way to know is for you to check (I know some sort of wang tiles support is in, but I don't know if it fits your use case)

**Comment by CosmicSpore** [comment]
I was reviewing some info on tilesets on the internet and randomly came across this ticket...
Anyway, I just wanted to point out that **Wang Corner tiles are ALREADY supported. That is what the 2x2 "bitmask mode" is.** It just isn't called that for some reason... The Wang tileset format can even be seen in an image on [the 'Tilemaps' tutorial page in the official Godot documentation](https://docs.godotengine.org/en/stable/tutorials/2d/using_tilemaps.html).

I was going to say that perhaps the TC meant Wang SIDE tiles... However, I'll also point out that this is ALSO supported, just a bit trickier to set up. 
**For Wang Side tiles, you use the 3x3 minimal "bitmask mode" and have to set the "ignore flag" (shift+click)** on a few of the corners. [This is ALSO stated in the 'Tilemaps' tutorial page within the documentation, with image.](https://docs.godotengine.org/en/stable/tutorials/2d/using_tilemaps.html)
The format image and instructions are all right underneath the title "3x3 minimal" and subtitle: "Template - Generic 16 tiles"

So there is really no need for this ticket except to point out that Godot (and the documentation) is rather unclear and should more clearly state that these setups are specifically called "Wang tiles", and probably also mention the full '3x3 minimal' setup are called "Blob" tiles. (I believe these are their official names for the formats, "Wang" being named after the creator of the base concept. I'm not sure why 'Blob tiles' are called "Blob", but it seems to be a fairly standard name for them when using various software or looking them up online.)

Anyway, I hope all this information helps someone!
